### PR TITLE
Clone fid experiment

### PR DESCRIPTION
**Pull request by ajay-paratmandali**
- Perform clone constrain related experiment like
    - colocation
    - location
    - order
  - Find multiple clone options and its meaning.
  - Find multiple way to configure clone resource having multiple instances on node.
  - Explore Group clone.

### CODE REVIEW

File: Pacemaker/clone_experiments/clone_resource.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    Create 11 resource of s3server with group s3server_group and then clone that group.
+    ```
+  - We can use clone group property for io-stack and management stack.
+  - resource-discovery can be used to reduce network traffic in large cluster.
+
+# constraint
+  - location:
+    - prefers, avoid for normal case like allow/not allow resource on current node even if other node not available.
+    - complex: `constraint location csm-kibana rule score=-INFINITY '#uname' eq $lnode and consul-c1-running eq 0`
+    - Use **resource-discovery**=never if we don't want resource to run anywhere in system.
```
Comment (indrajitzagade) [inline Pacemaker/clone_experiments/clone_resource.md:37]: "if we don't want resource to run anywhere in system" I am not sure if this statement is true. Please confirm.

File: Pacemaker/clone_experiments/clone_resource.md [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    $ pcs resource create s3server-9 ocf:seagate:clone-service service=s3server --group s3server
+    $ pcs resource create s3server-10 ocf:seagate:clone-service service=s3server --group s3server
+    $ pcs resource create s3server-11 ocf:seagate:clone-service service=s3server --group s3server
+    $ pcs resource clone s3server
+
+    Clone Set: s3server-clone [s3server]
+      Started: [ srvnode-1 srvnode-2 srvnode-3 ]
+```
+
+# Clone Group
```
Comment (indrajitzagade) [inline Pacemaker/clone_experiments/clone_resource.md:307]: Is this the recommended approach for LDR_R2? If not, can we please document the recommendations coming out of this POC?

File: Pacemaker/clone_experiments/clone_resource.md (hunk 1 above)
Comment (ajay-paratmandali) [reply Pacemaker/clone_experiments/clone_resource.md]: Modified description its already given below. It was like "Pacemaker will not discover resource on this node."

File: Pacemaker/clone_experiments/clone_resource.md (hunk 2 above)
Comment (ajay-paratmandali) [reply Pacemaker/clone_experiments/clone_resource.md:307]: Here i only done exploration of resource group and when to use it. currently approach is not finalize.

File: Pacemaker/clone_experiments/clone_resource.md (hunk 1 above)
Comment (indrajitzagade) [reply Pacemaker/clone_experiments/clone_resource.md]: ok

File: Pacemaker/clone_experiments/clone_resource.md (hunk 2 above)
Comment (indrajitzagade) [reply Pacemaker/clone_experiments/clone_resource.md:307]: ok

**Review by indrajitzagade — APPROVED** [review]

**Review by ajaysrivas — APPROVED** [review]

**Review by azheregelya — CHANGES_REQUESTED** [review]

File: Pacemaker/clone_experiments/clone_resource.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+     Stopped: [ srvnode-1 srvnode-2 srvnode-3 ]
+  Clone Set: statsd-clone [statsd]
+     Started: [ srvnode-3 ]
+     Stopped: [ srvnode-1 srvnode-2 ]
+  ```
+
+  4. Stop start clone on particular node
+  ```
+  pcs resource ban statsd-clone srvnode-3
+  pcs resource clear statsd-clone
```
[the remark this answers is not in the document]
Comment (ajay-paratmandali) [reply Pacemaker/clone_experiments/clone_resource.md:179]: `clear will remove constraints created by move and/or ban on the specified resource. Due to this resource can start on node if it was banned.`

Added info in doc also

File: Pacemaker/clone_experiments/clone_resource.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    - If the value of this option is true, a copy of the clone running on one machine is not equivalent to another instance, whether that instance is running on another node or on the same node. The default value is true if the value of clone-node-max is greater than one; otherwise the default value is false.
+
+  - ordered
+    - Should the copies be started in series (instead of in parallel). Allowed values: false, true. The default value is false.
+
+  - interleave
+    - Changes the behavior of ordering constraints (between clones/masters) so that copies of the first clone can start or stop as soon as the copy on the same node of the second clone has started or stopped (rather than waiting until every instance of the second clone has started or stopped). Allowed values: false, true. The default value is false.
+
+### Test case for constrain and clone options.
+  1. colocation of clone and unclone resource
```
[the remark this answers is not in the document]
Comment (ajay-paratmandali) [reply Pacemaker/clone_experiments/clone_resource.md]: it is active_passive resource with single clone 

in doc changed word to active_passive resource.

File: Pacemaker/clone_experiments/clone_resource.md [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+     s3server:28        (ocf::seagate:s3server):        Started srvnode-1
+     s3server:29        (ocf::seagate:s3server):        Started srvnode-2
+     s3server:30        (ocf::seagate:s3server):        Started srvnode-3
+     s3server:31        (ocf::seagate:s3server):        Started srvnode-1
+     s3server:32        (ocf::seagate:s3server):        Started srvnode-2
+```
+
+### Approach 2: Create resource per node with max clone size per node.
+  - Here each node have s3server resource having 11 clone.
+  - Approach good if clone if node less and clone can not move across node.
```
[the remark this answers is not in the document]
Comment (ajay-paratmandali) [reply Pacemaker/clone_experiments/clone_resource.md]: Done.

**Comment by ajay-paratmandali** [comment]
@azheregelya 
addressed comment please review.

**Pull request merged**